File: backend/app/schemas/provider_usage.py
from datetime import date, datetime

from pydantic import BaseModel


class ProviderUsageItem(BaseModel):
    provider: str
    label: str
    quota_limit: int | None
    request_count: int
    success_count: int
    error_count: int
    remaining_count: int | None
    usage_month: date
    last_status: str | None = None
    last_used_at: datetime | None = None
    message: str


class ProviderUsageResponse(BaseModel):
    items: list[ProviderUsageItem]

File: frontend/src/types/provider-usage.ts
export type ProviderUsageItem = {
  provider: string;
  label: string;
  quota_limit: number | null;
  request_count: number;
  success_count: number;
  error_count: number;
  remaining_count: number | null;
  usage_month: string;
  last_status: string | null;
  last_used_at: string | null;
  message: string;
};

export type ProviderUsageResponse = {
  items: ProviderUsageItem[];
};
